dough-inventories/src/main/java/io/github/bakedlibs/dough/inventory/Menu.java

```diff
@@ -0,0 +1,86 @@
+package io.github.bakedlibs.dough.inventory;
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.InventoryView;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.bakedlibs.dough.inventory.factory.MenuFactory;
+
+public interface Menu extends InventoryHolder {
+
+    /**
+     * This method returns the {@link MenuFactory} which was used
+     * to create this {@link Menu}.
+     * 
+     * @return The original {@link MenuFactory}
+     */
+    @Nonnull
+    MenuFactory getFactory();
+
+    /**
+     * This returns the {@link MenuLayout} which was used to create
+     * this {@link Menu}.
+     * 
+     * @return The {@link MenuLayout}
+     */
+    @Nonnull
+    MenuLayout getLayout();
+
+    /**
+     * This returns the title of this {@link Menu}.
+     * If no title was set, null will be returned.
+     * 
+     * @return The title of this {@link Menu} or null
+     */
+    @Nullable
+    String getTitle();
+
+    void setAll(@Nonnull SlotGroup group, @Nullable ItemStack item);
```

Comment:
Why no docs for the methods from now on up until the last?

Reply:
Documentation is still incomplete, that is why